### PR TITLE
Update github.com/thanos-io/objstore

### PR DESCRIPTION
**Pull request by simonswine**
This updates objstore to the latest version. My objective is to get the latest metrics on that library into grafana/pyroscope. As we are currently depending on mimir we also need to upgrade the mimir code base.

There have been a few breaking changes in how buckets get instrumented for tracing/metrics, so it requires this PR.

### CODE REVIEW

File: pkg/storegateway/bucket.go [hunk 1]
```diff
@@ -611,12 +611,12 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
...
 	// but sometimes it can take minutes if the block isn't loaded and there is a surge in queries for unloaded blocks.
-	tracing.DoWithSpan(ctx, "store_query_gate_ismyturn", func(ctx context.Context, _ tracing.Span) {
-		err = s.queryGate.Start(ctx)
-	})
+	span, spanCtx := opentracing.StartSpanFromContext(ctx, "store_query_gate_ismyturn")
+	err = s.queryGate.Start(spanCtx)
 	if err != nil {
 		return errors.Wrapf(err, "failed to wait for turn")
 	}
+	span.Finish()
```
Comment (pstibrany) [inline pkg/storegateway/bucket.go:619]: This will not finish span in case of error. We should move `span.Finish()` before error check.

**Review by pstibrany — COMMENTED** [review]
Thank you. LGTM overall, with a tiny bug.

File: pkg/storegateway/bucket.go (hunk 1 above)
Comment (simonswine) [reply pkg/storegateway/bucket.go]: Oh yes, you are totally right. Well spotted and fixed in f45fb5ab7de643cdfc38746b3234634c82656990

**Review by pstibrany — APPROVED** [review]
Thank you!

**Pull request merged**